src/parser.rs

```diff
@@ -2272,11 +2276,32 @@ impl<'a> Parser<'a> {
 
     /// Parse an INSERT statement
     pub fn parse_insert(&mut self) -> Result<Statement, ParserError> {
+        let or = if dialect_of!(self is SQLiteDialect) {
```

Comment:
This can be reverted / simplified with early return so it doesn't need double nesting, something like:
```rust
if ! dialect_of!(self is SQLiteDialect)  {
   None
...

```